# This file is part of HiGFXback

# requires
REQUIRES="bison-build elfutils-build findutils-build libstdc++-build make-build ncurses-build openssl-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

# configure
make `uname -m`_defconfig fragment-`uname -m`.config

# build
make

# install
install -d $DESTDIR/boot
install arch/`uname -m`/boot/bzImage $DESTDIR/boot

# build.pc
install -d $DESTDIR/share/pkgconfig
cat > $DESTDIR/share/pkgconfig/linux-build.pc << EOF
Name: linux
Version: 5.15
Description: Linux kernel
Requires: $REQUIRES

exec=\\
/boot/bzImage
EOF

exit

# patch
--- linux-5.15.orig/arch/x86/configs/fragment-x86_64.config
+++ linux-5.15/arch/x86/configs/fragment-x86_64.config
@@ -0,0 +1,10 @@
+CONFIG_SYSFB_SIMPLEFB=y
+CONFIG_INPUT_MOUSEDEV=y
+CONFIG_INPUT_UINPUT=y
+CONFIG_DRM_SIMPLEDRM=y
+CONFIG_FB=y
+CONFIG_FB_VESA=y
+CONFIG_SND_OSSEMUL=y
+CONFIG_SND_MIXER_OSS=y
+CONFIG_SND_PCM_OSS=y
+CONFIG_SND_HDA_CODEC_REALTEK=y
--- linux-5.15.orig/drivers/video/fbdev/core/fbcon.c
+++ linux-5.15/drivers/video/fbdev/core/fbcon.c
@@ -2079,6 +2079,7 @@
 	 */
 	info->var.activate = var.activate;
 	var.vmode |= info->var.vmode & ~FB_VMODE_MASK;
+	memcpy(var.reserved, info->var.reserved, sizeof(var.reserved));
 	fb_set_var(info, &var);
 	ops->var = info->var;
 
# source
https://kernel.org/pub/linux/kernel/v5.x/linux-5.15.tar.xz
